src/amchart/CommonFunnel.js

```diff
@@ -77,10 +81,13 @@
         if(this.reverseDataSorting()){
             sortingMethod = function(a,b){ return a[1] < b[1] ? 1 : -1; };
         }
-        this.data(this.data().sort(sortingMethod));
+        this.data().sort(sortingMethod);
 
         // DataProvider
-        this._chart.dataProvider = this.formatData(this.data());
+        if (this._dataUpdated > this._lastDataUpdated) {
+            this._chart.dataProvider = this.formatData(this.data());
```

Comment:
would be nifty to maybe use the d3 array adapter to "monitor" for changes in data ... not sure if im thinking right about that though @GordonSmith thoughts?